src/opm/parser/eclipse/Parser/raw/RawKeyword.hpp

```diff
@@ -0,0 +1,80 @@
+/*
+  Copyright 2013 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef RAWKEYWORD_HPP
+#define RAWKEYWORD_HPP
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <list>
+
+#include "RawEnums.hpp"
+#include "RawConsts.hpp"
+
+namespace Opm {
+
+    class RawRecord;
+    class string_view;
+
+    class RawKeyword {
+    public:
+        RawKeyword(const std::string& name, const std::string& filename, std::size_t lineNR, bool raw_string, Raw::KeywordSizeEnum sizeType);
+        RawKeyword(const std::string& name, const std::string& filename, std::size_t lineNR, bool raw_string, Raw::KeywordSizeEnum sizeType, std::size_t size_arg);
+        bool terminateKeyword();
+        bool addRecord(RawRecord record);
+
+        const std::string& getKeywordName() const;
+        Raw::KeywordSizeEnum getSizeType() const;
+
+        // Special case method only for inspecting INCLUDE keywords;
+        // the general getRecords functionality should use the
+        // iterator interface.
+        const RawRecord& getFirstRecord( ) const;
+
+        bool isFinished() const;
+        bool unKnownSize() const;
+        bool rawStringKeyword() const;
+        std::pair<std::string, std::size_t> getLocation() const;
+
+        using const_iterator = std::list< RawRecord >::const_iterator;
+        using iterator = std::list< RawRecord >::iterator;
+
+        iterator begin();
+        iterator end();
+        const_iterator begin() const;
+        const_iterator end() const;
+    private:
+        std::string m_name;
+        std::string m_filename;
+
+        size_t m_lineNR;
+        bool raw_string_keyword;
+        Raw::KeywordSizeEnum m_sizeType;
+
+        size_t m_fixedSize = 0;
+        size_t m_numTables = 0;
+        size_t m_currentNumTables = 0;
+        bool m_isFinished = false;
+
+        std::list< RawRecord > m_records;
```

Comment:
Has anyone benchmarked this to verify that `std::list` is a good representation for this data member?

Reply:
I have not benchmarked it - and doubt  anyone else has done. I don't know why `std::list` was chosen here, and it could/should probably be changed to `std::vector`; but it has worked for ~6 years - so I guess it is ok.